content/docs/user-guide/dvc-files-and-directories.md

```diff
@@ -1,8 +1,142 @@
+  when `dvc import` and `dvc import-url` are used. Typically there is only one
+  (but several can be added manually).
+- `meta` (optional): Arbitrary metadata can be added manually with this field.
+  Any YAML contents is supported. `meta` contents are ignored by DVC, but they
+  can be meaningful for user processes that read `.dvc` files.
+
+An _output entry_ can consist of these fields:
+
+- `md5`: Hash value for the output file
+- `path`: Path to the output in the <abbr>workspace</abbr>, relative to the
+  location of the `.dvc` file
+- `cache`: Whether or not DVC should cache the output. `true` by default
+
+A _dependency entry_ consists of a these possible fields:
+
+- `path`: Path to the dependency, relative to the `wdir` path (always present)
+- `md5`: MD5 hash for the dependency (most [stages](/doc/command-reference/run))
+- `etag`: Strong ETag response header (only HTTP <abbr>external
+  dependencies</abbr> created with `dvc import-url`)
+- `repo`: This entry is only for external dependencies created with
+  `dvc import`, and can contains the following fields:
+
+  - `url`: URL of Git repository with source DVC project
+  - `rev`: Only present when the `--rev` option of `dvc import` is used.
+    Specific commit hash, branch or tag name, etc. (a
+    [Git revision](https://git-scm.com/docs/revisions)) used to import the
+    dependency from.
+  - `rev_lock`: Git commit hash of the external <abbr>DVC repository</abbr> at
+    the time of importing or updating (with `dvc update`) the dependency.
+
+Note that comments can be added to `.dvc` files and `dvc.yaml` using the
+`# comment` syntax. `meta` fields and `#` comments are preserved among
+executions of the `dvc repro` and `dvc commit` commands, but not when a `.dvc`
+file is overwritten by `dvc add`,`dvc import`, or `dvc import-url`.
+
+## dvc.yaml files
+
+When you add commands to a pipeline with `dvc run`, the `dvc.yaml` file is
+created or updated. Here's a simple example:
+
+```yaml
+stages:
+  features:
+    cmd: jupyter nbconvert --execute featurize.ipynb
+    deps:
+      - data/clean
+    params:
+      - levels.no
+    outs:
+      - features
+    metrics:
+      - performance.json
+  training:
+    cmd: python train.py
+    deps:
+      - train.py
+      - features
+    outs:
+      - model.pkl
+    plots:
+      - logs.csv:
+          x: epoch
+          x_label: Epoch
+    meta: 'For deployment'
+    # User metadata and comments are supported.
+```
+
+`dvc.yaml` files consists of a group of `stages` with names provided explicitly
+by the user with the `--name` (`-n`) option of `dvc run`. Each stage can contain
+the possible following fields:
+
+- `cmd` (always present): Executable command defined in this stage
+- `deps`: List of <abbr>dependency</abbr> file or directory paths of this stage
+- `params`: List of the [parameters](/doc/command-reference/params). These are
+  key paths referring to another YAML file (`params.yaml` by default).
```

Comment:
only YAML is supported? or JSON as well? just to double check

Reply:
Good catch, also JSON indeed according to https://dvc.org/doc/command-reference/params#examples. It's just not something I've ever tried or was very aware of. Updating...